src/Aspire.Hosting.Azure.SignalR/api/Aspire.Hosting.Azure.SignalR.cs

```diff
@@ -10,18 +10,40 @@ namespace Aspire.Hosting
 {
     public static partial class AzureSignalRExtensions
     {
+        public static ApplicationModel.IResourceBuilder<ApplicationModel.AzureSignalRResource> AddAzureSignalR(this IDistributedApplicationBuilder builder, string name, ApplicationModel.AzureSignalRServiceMode serviceMode) { throw null; }
+
         public static ApplicationModel.IResourceBuilder<ApplicationModel.AzureSignalRResource> AddAzureSignalR(this IDistributedApplicationBuilder builder, string name) { throw null; }
+
+        public static ApplicationModel.IResourceBuilder<ApplicationModel.AzureSignalRResource> RunAsEmulator(this ApplicationModel.IResourceBuilder<ApplicationModel.AzureSignalRResource> builder, System.Action<ApplicationModel.IResourceBuilder<Azure.AzureSignalREmulatorResource>>? configureContainer = null) { throw null; }
     }
 }
 
 namespace Aspire.Hosting.ApplicationModel
 {
-    public partial class AzureSignalRResource : Azure.AzureProvisioningResource, IResourceWithConnectionString, IResource, IManifestExpressionProvider, IValueProvider, IValueWithReferences
+    public partial class AzureSignalRResource : Azure.AzureProvisioningResource, IResourceWithConnectionString, IResource, IManifestExpressionProvider, IValueProvider, IValueWithReferences, IResourceWithEndpoints
```

Comment:
I'm not sure this should implement `IResourceWithEndpoints`. It might not be the end of the world, but we don't do this for other azure resources that expose emulators.

Reply:
Nope, I'm crazy this is fine.